From f93dc8098dbeb15b722a5c6b643e4e666e047ade Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 3 Mar 2025 12:20:21 +0800
Subject: [PATCH 104/104] libweston: bindings: Disable extra bindings by
 default

Set env WESTON_MOD_BINDING WESTON_KEY_BINDING WESTON_DEBUG_BINDING to
enable.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/bindings.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/libweston/bindings.c b/libweston/bindings.c
index 4d2ad94f6..3392efe64 100644
--- a/libweston/bindings.c
+++ b/libweston/bindings.c
@@ -52,6 +52,9 @@ weston_compositor_add_binding(struct weston_compositor *compositor,
 {
 	struct weston_binding *binding;
 
+	if (!getenv("WESTON_MOD_BINDING") && modifier)
+		return NULL;
+
 	binding = malloc(sizeof *binding);
 	if (binding == NULL)
 		return NULL;
@@ -75,6 +78,9 @@ weston_compositor_add_key_binding(struct weston_compositor *compositor,
 {
 	struct weston_binding *binding;
 
+	if (!getenv("WESTON_KEY_BINDING"))
+		return NULL;
+
 	binding = weston_compositor_add_binding(compositor, key, 0, 0,
 						modifier, handler, data);
 	if (binding == NULL)
@@ -186,8 +192,13 @@ weston_compositor_add_debug_binding(struct weston_compositor *compositor,
 {
 	struct weston_binding *binding;
 
+	if (!getenv("WESTON_DEBUG_BINDING"))
+		return NULL;
+
 	binding = weston_compositor_add_binding(compositor, key, 0, 0, 0,
 						handler, data);
+	if (binding == NULL)
+		return NULL;
 
 	wl_list_insert(compositor->debug_binding_list.prev, &binding->link);
 
-- 
2.20.1

